aptos-move/parallel-executor/src/scheduler.rs

```diff
@@ -416,16 +455,36 @@ impl Scheduler {
         // If successfully incarnated (changed status from ready to executing),
         // return version and guard for execution task, otherwise None.
         self.try_incarnate(idx_to_execute)
-            .map(|incarnation| ((idx_to_execute, incarnation), guard))
+            .map(|(incarnation, maybe_condvar)| {
+                if incarnation == 0 {
+                    // counting transactions that ever got scheduled for execution.
+                    self.drain_idx.fetch_add(1, Ordering::SeqCst);
```

Comment:
do we still need this?

Reply:
This we still do, otherwise, when stop_idx is updated, any execution task suspended on a higher index will become a zombie and never let anyone return.